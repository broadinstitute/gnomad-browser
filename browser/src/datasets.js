export const referenceGenomeForDataset = datasetId => {
  if (datasetId.startsWith('gnomad_r3')) {
    return 'GRCh38'
  }

  return 'GRCh37'
}

export const datasetLabels = {
  exac: 'ExAC v1.0',
  gnomad_r2_1: 'gnomAD v2.1.1',
  gnomad_r2_1_controls: 'gnomAD v2.1.1 (controls)',
  gnomad_r2_1_non_cancer: 'gnomAD v2.1.1 (non-cancer)',
  gnomad_r2_1_non_neuro: 'gnomAD v2.1.1 (non-neuro)',
  gnomad_r2_1_non_topmed: 'gnomAD v2.1.1 (non-TOPMed)',
  gnomad_r3: 'gnomAD v3.1.2',
  gnomad_r3_controls_and_biobanks: 'gnomAD v3.1.2 (controls/biobanks)',
  gnomad_r3_non_cancer: 'gnomAD v3.1.2 (non-cancer)',
  gnomad_r3_non_neuro: 'gnomAD v3.1.2 (non-neuro)',
  gnomad_r3_non_topmed: 'gnomAD v3.1.2 (non-TOPMed)',
  gnomad_r3_non_v2: 'gnomAD v3.1.2 (non-v2)',
  gnomad_sv_r2_1: 'gnomAD SVs v2.1',
  gnomad_sv_r2_1_controls: 'gnomAD SVs v2.1 (controls)',
  gnomad_sv_r2_1_non_neuro: 'gnomAD SVs v2.1 (non-neuro)',
}

export const allDatasetIds = Object.keys(datasetLabels)

export const labelForDataset = datasetId => datasetLabels[datasetId] || 'Unknown'
