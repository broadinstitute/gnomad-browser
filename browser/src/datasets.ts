import { datasetLabels } from '../../dataset-metadata/metadata'

export const referenceGenomeForDataset = (datasetId: any) => {
  if (datasetId.startsWith('gnomad_r3')) {
    return 'GRCh38'
  }

  return 'GRCh37'
}

// @ts-expect-error TS(7053) FIXME: Element implicitly has an 'any' type because expre... Remove this comment to see the full error message
export const labelForDataset = (datasetId: any) => datasetLabels[datasetId] || 'Unknown'
