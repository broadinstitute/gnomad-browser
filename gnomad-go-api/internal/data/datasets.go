package data

type DatasetConfig struct {
	ID              string
	ReferenceGenome string
	HasExome        bool
	HasGenome       bool
	HasJoint        bool
}

var Datasets = map[string]DatasetConfig{
	"gnomad_r4": {
		ID:              "gnomad_r4",
		ReferenceGenome: "GRCh38",
		HasExome:        true,
		HasGenome:       true,
		HasJoint:        true,
	},
	"gnomad_r4_non_ukb": {
		ID:              "gnomad_r4_non_ukb",
		ReferenceGenome: "GRCh38",
		HasExome:        true,
		HasGenome:       true,
		HasJoint:        true,
	},
	"gnomad_r3": {
		ID:              "gnomad_r3",
		ReferenceGenome: "GRCh38",
		HasExome:        false,
		HasGenome:       true,
		HasJoint:        false,
	},
	"gnomad_r3_controls_and_biobanks": {
		ID:              "gnomad_r3_controls_and_biobanks",
		ReferenceGenome: "GRCh38",
		HasExome:        false,
		HasGenome:       true,
		HasJoint:        false,
	},
	"gnomad_r3_non_cancer": {
		ID:              "gnomad_r3_non_cancer",
		ReferenceGenome: "GRCh38",
		HasExome:        false,
		HasGenome:       true,
		HasJoint:        false,
	},
	"gnomad_r3_non_neuro": {
		ID:              "gnomad_r3_non_neuro",
		ReferenceGenome: "GRCh38",
		HasExome:        false,
		HasGenome:       true,
		HasJoint:        false,
	},
	"gnomad_r3_non_topmed": {
		ID:              "gnomad_r3_non_topmed",
		ReferenceGenome: "GRCh38",
		HasExome:        false,
		HasGenome:       true,
		HasJoint:        false,
	},
	"gnomad_r3_non_v2": {
		ID:              "gnomad_r3_non_v2",
		ReferenceGenome: "GRCh38",
		HasExome:        false,
		HasGenome:       true,
		HasJoint:        false,
	},
	"gnomad_r2_1": {
		ID:              "gnomad_r2_1",
		ReferenceGenome: "GRCh37",
		HasExome:        true,
		HasGenome:       true,
		HasJoint:        false,
	},
	"gnomad_r2_1_controls": {
		ID:              "gnomad_r2_1_controls",
		ReferenceGenome: "GRCh37",
		HasExome:        true,
		HasGenome:       true,
		HasJoint:        false,
	},
	"gnomad_r2_1_non_cancer": {
		ID:              "gnomad_r2_1_non_cancer",
		ReferenceGenome: "GRCh37",
		HasExome:        true,
		HasGenome:       true,
		HasJoint:        false,
	},
	"gnomad_r2_1_non_neuro": {
		ID:              "gnomad_r2_1_non_neuro",
		ReferenceGenome: "GRCh37",
		HasExome:        true,
		HasGenome:       true,
		HasJoint:        false,
	},
	"gnomad_r2_1_non_topmed": {
		ID:              "gnomad_r2_1_non_topmed",
		ReferenceGenome: "GRCh37",
		HasExome:        true,
		HasGenome:       true,
		HasJoint:        false,
	},
	"exac": {
		ID:              "exac",
		ReferenceGenome: "GRCh37",
		HasExome:        true,
		HasGenome:       false,
		HasJoint:        false,
	},
	"gnomad_sv_r2_1": {
		ID:              "gnomad_sv_r2_1",
		ReferenceGenome: "GRCh37",
		HasExome:        false,
		HasGenome:       false,
		HasJoint:        false,
	},
	"gnomad_sv_r2_1_controls": {
		ID:              "gnomad_sv_r2_1_controls",
		ReferenceGenome: "GRCh37",
		HasExome:        false,
		HasGenome:       false,
		HasJoint:        false,
	},
	"gnomad_sv_r2_1_non_neuro": {
		ID:              "gnomad_sv_r2_1_non_neuro",
		ReferenceGenome: "GRCh37",
		HasExome:        false,
		HasGenome:       false,
		HasJoint:        false,
	},
	"gnomad_sv_r4": {
		ID:              "gnomad_sv_r4",
		ReferenceGenome: "GRCh38",
		HasExome:        false,
		HasGenome:       false,
		HasJoint:        false,
	},
	"gnomad_cnv_r4": {
		ID:              "gnomad_cnv_r4",
		ReferenceGenome: "GRCh38",
		HasExome:        false,
		HasGenome:       false,
		HasJoint:        false,
	},
}
