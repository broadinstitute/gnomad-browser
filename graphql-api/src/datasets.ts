export const DATASET_LABELS = {
  gnomad_r4: 'gnomAD v4',
  gnomad_cnv_r4: 'gnomAD v4 cnv',
  gnomad_r3: 'gnomAD v3',
  gnomad_r3_controls_and_biobanks: 'gnomAD v3 (controls and biobanks)',
  gnomad_r3_non_cancer: 'gnomAD v3 (non-cancer)',
  gnomad_r3_non_neuro: 'gnomAD v3 (non-neuro)',
  gnomad_r3_non_topmed: 'gnomAD v3 (non-TOPMed)',
  gnomad_r3_non_v2: 'gnomAD v3 (non-v2)',
  gnomad_r2_1: 'gnomAD v2',
  gnomad_r2_1_controls: 'gnomAD v2 (controls)',
  gnomad_r2_1_non_neuro: 'gnomAD v2 (non-neuro)',
  gnomad_r2_1_non_cancer: 'gnomAD v2 (non-cancer)',
  gnomad_r2_1_non_topmed: 'gnomAD v2 (non-TOPMed)',
  exac: 'ExAC',
}

export const DATASET_REFERENCE_GENOMES: Record<string, string> = {
  gnomad_r4: 'GRCh38',
  gnomad_cnv_r4: 'GRCh38',
  gnomad_r3: 'GRCh38',
  gnomad_r3_controls_and_biobanks: 'GRCh38',
  gnomad_r3_non_cancer: 'GRCh38',
  gnomad_r3_non_neuro: 'GRCh38',
  gnomad_r3_non_topmed: 'GRCh38',
  gnomad_r3_non_v2: 'GRCh38',
  gnomad_r2_1: 'GRCh37',
  gnomad_r2_1_controls: 'GRCh37',
  gnomad_r2_1_non_neuro: 'GRCh37',
  gnomad_r2_1_non_cancer: 'GRCh37',
  gnomad_r2_1_non_topmed: 'GRCh37',
  exac: 'GRCh37',
}
