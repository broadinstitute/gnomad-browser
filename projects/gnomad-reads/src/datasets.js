import logger from './logging'

const datasets = (process.env.DATASETS || '')
  .split(',')
  .filter(Boolean)
  .reduce(
    (acc, datasetId) => ({
      ...acc,
      [datasetId.toLowerCase()]: {
        exomes: {
          readsDirectory: process.env[`${datasetId.toUpperCase()}_EXOMES_READS_DIRECTORY`],
          publicPath: process.env[`${datasetId.toUpperCase()}_EXOMES_PUBLIC_PATH`],
        },
        genomes: {
          readsDirectory: process.env[`${datasetId.toUpperCase()}_GENOMES_READS_DIRECTORY`],
          publicPath: process.env[`${datasetId.toUpperCase()}_GENOMES_PUBLIC_PATH`],
        },
      },
    }),
    {}
  )

logger.info(`Dataset configuration: ${JSON.stringify(datasets)}`)

export default datasets
