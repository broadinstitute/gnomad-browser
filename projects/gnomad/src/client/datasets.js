export const referenceGenomeForDataset = datasetId => {
  if (datasetId === 'gnomad_r3') {
    return 'GRCh38'
  }

  return 'GRCh37'
}

const datasetLabels = {
  exac: 'ExAC v1.0',
  gnomad_r2_1: 'gnomAD v2.1.1',
  gnomad_r2_1_controls: 'gnomAD v2.1.1 (controls)',
  gnomad_r2_1_non_cancer: 'gnomAD v2.1.1 (non-cancer)',
  gnomad_r2_1_non_neuro: 'gnomAD v2.1.1 (non-neuro)',
  gnomad_r2_1_non_topmed: 'gnomAD v2.1.1 (non-TOPMed)',
  gnomad_r3: 'gnomAD v3',
  gnomad_sv_r2_1: 'gnomAD SVs v2.1',
  gnomad_sv_r2_1_controls: 'gnomAD SVs v2.1 (controls)',
  gnomad_sv_r2_1_non_neuro: 'gnomAD SVs v2.1 (non-neuro)',
}

export const labelForDataset = datasetId => datasetLabels[datasetId] || 'Unknown'
