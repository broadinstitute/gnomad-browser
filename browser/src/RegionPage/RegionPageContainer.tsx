import React from 'react'

import { isRegionId, parseRegionId } from '@gnomad/identifiers'
import { Page, PageHeading } from '@gnomad/ui'

import { DatasetId } from '../../../dataset-metadata/metadata'
import { referenceGenomeForDataset } from '../datasets'
import DocumentTitle from '../DocumentTitle'
import Query from '../Query'
import RegionPage from './RegionPage'

const operationName = 'Region'
const query = `
  query ${operationName}($chrom: String!, $start: Int!, $stop: Int!, $referenceGenome: ReferenceGenomeId!, $shortTandemRepeatDatasetId: DatasetId!, $includeShortTandemRepeats: Boolean!) {
    region(chrom: $chrom, start: $start, stop: $stop, reference_genome: $referenceGenome) {
      genes {
        gene_id
        symbol
        start
        stop
        exons {
          feature_type
          start
          stop
        }
        transcripts {
          transcript_id
          exons {
            feature_type
            start
            stop
          }
        }
      }
      short_tandem_repeats(dataset: $shortTandemRepeatDatasetId) @include(if: $includeShortTandemRepeats) {
        id
      }
    }
  }
`

type Props = {
  datasetId: DatasetId
  regionId: string
}

const RegionPageContainer = ({ datasetId, regionId }: Props) => {
  if (!isRegionId(regionId)) {
    return (
      // @ts-expect-error TS(2746) FIXME: This JSX tag's 'children' prop expects a single ch... Remove this comment to see the full error message
      <Page>
        <DocumentTitle title="Invalid region ID" />
        <PageHeading>Invalid region ID</PageHeading>
        <p>Regions IDs must be formatted chrom-start-stop.</p>
      </Page>
    )
  }

  const { chrom, start, stop } = parseRegionId(regionId)
  const referenceGenome = referenceGenomeForDataset(datasetId)

  return (
    <Query
      operationName={operationName}
      query={query}
      variables={{
        chrom,
        start,
        stop,
        referenceGenome,
        includeShortTandemRepeats: datasetId.startsWith('gnomad_r3'),
        shortTandemRepeatDatasetId: 'gnomad_r3',
      }}
      loadingMessage="Loading region"
      errorMessage="Unable to load region"
      success={(data: any) => data.region}
    >
      {({ data }: any) => {
        return (
          <RegionPage
            datasetId={datasetId}
            region={{
              ...data.region,
              reference_genome: referenceGenome,
              chrom: chrom === 'MT' ? 'M' : chrom,
              start,
              stop,
            }}
          />
        )
      }}
    </Query>
  )
}

export default RegionPageContainer
