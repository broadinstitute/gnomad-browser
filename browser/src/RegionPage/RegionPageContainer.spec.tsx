import React from 'react'
import { test, expect } from '@jest/globals'
import { createRenderer } from 'react-test-renderer/shallow'

import { forAllDatasets } from '../../../tests/__helpers__/datasets'
import RegionPageContainer from './RegionPageContainer'

forAllDatasets('RegionPageContainer with dataset %s', (datasetId) => {
  test('queries API with correct parameters', () => {
    const renderer = createRenderer()
    renderer.render(<RegionPageContainer datasetId={datasetId} regionId="12-345-678" />)
    expect(renderer.getRenderOutput()).toMatchSnapshot()
  })
})
