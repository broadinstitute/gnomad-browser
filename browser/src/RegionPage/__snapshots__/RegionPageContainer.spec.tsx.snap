// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegionPageContainer with dataset exac queries API with correct parameters 1`] = `
<Query
  errorMessage="Unable to load region"
  loadingMessage="Loading region"
  operationName="Region"
  query="
  query Region($chrom: String!, $start: Int!, $stop: Int!, $referenceGenome: ReferenceGenomeId!, $shortTandemRepeatDatasetId: DatasetId!, $includeShortTandemRepeats: Boolean!) {
    region(chrom: $chrom, start: $start, stop: $stop, reference_genome: $referenceGenome) {
      genes {
        gene_id
        symbol
        start
        stop
        exons {
          feature_type
          start
          stop
        }
        transcripts {
          transcript_id
          exons {
            feature_type
            start
            stop
          }
        }
      }
      non_coding_constraints {
        start
        stop
        oe
        z
      }
      short_tandem_repeats(dataset: $shortTandemRepeatDatasetId) @include(if: $includeShortTandemRepeats) {
        id
      }
    }
  }
"
  success={[Function]}
  url="/api/"
  variables={
    {
      "chrom": "12",
      "includeShortTandemRepeats": false,
      "referenceGenome": "GRCh37",
      "shortTandemRepeatDatasetId": "gnomad_r3",
      "start": 345,
      "stop": 678,
    }
  }
>
  [Function]
</Query>
`;

exports[`RegionPageContainer with dataset gnomad_r2_1 queries API with correct parameters 1`] = `
<Query
  errorMessage="Unable to load region"
  loadingMessage="Loading region"
  operationName="Region"
  query="
  query Region($chrom: String!, $start: Int!, $stop: Int!, $referenceGenome: ReferenceGenomeId!, $shortTandemRepeatDatasetId: DatasetId!, $includeShortTandemRepeats: Boolean!) {
    region(chrom: $chrom, start: $start, stop: $stop, reference_genome: $referenceGenome) {
      genes {
        gene_id
        symbol
        start
        stop
        exons {
          feature_type
          start
          stop
        }
        transcripts {
          transcript_id
          exons {
            feature_type
            start
            stop
          }
        }
      }
      non_coding_constraints {
        start
        stop
        oe
        z
      }
      short_tandem_repeats(dataset: $shortTandemRepeatDatasetId) @include(if: $includeShortTandemRepeats) {
        id
      }
    }
  }
"
  success={[Function]}
  url="/api/"
  variables={
    {
      "chrom": "12",
      "includeShortTandemRepeats": false,
      "referenceGenome": "GRCh37",
      "shortTandemRepeatDatasetId": "gnomad_r3",
      "start": 345,
      "stop": 678,
    }
  }
>
  [Function]
</Query>
`;

exports[`RegionPageContainer with dataset gnomad_r2_1_controls queries API with correct parameters 1`] = `
<Query
  errorMessage="Unable to load region"
  loadingMessage="Loading region"
  operationName="Region"
  query="
  query Region($chrom: String!, $start: Int!, $stop: Int!, $referenceGenome: ReferenceGenomeId!, $shortTandemRepeatDatasetId: DatasetId!, $includeShortTandemRepeats: Boolean!) {
    region(chrom: $chrom, start: $start, stop: $stop, reference_genome: $referenceGenome) {
      genes {
        gene_id
        symbol
        start
        stop
        exons {
          feature_type
          start
          stop
        }
        transcripts {
          transcript_id
          exons {
            feature_type
            start
            stop
          }
        }
      }
      non_coding_constraints {
        start
        stop
        oe
        z
      }
      short_tandem_repeats(dataset: $shortTandemRepeatDatasetId) @include(if: $includeShortTandemRepeats) {
        id
      }
    }
  }
"
  success={[Function]}
  url="/api/"
  variables={
    {
      "chrom": "12",
      "includeShortTandemRepeats": false,
      "referenceGenome": "GRCh37",
      "shortTandemRepeatDatasetId": "gnomad_r3",
      "start": 345,
      "stop": 678,
    }
  }
>
  [Function]
</Query>
`;

exports[`RegionPageContainer with dataset gnomad_r2_1_non_cancer queries API with correct parameters 1`] = `
<Query
  errorMessage="Unable to load region"
  loadingMessage="Loading region"
  operationName="Region"
  query="
  query Region($chrom: String!, $start: Int!, $stop: Int!, $referenceGenome: ReferenceGenomeId!, $shortTandemRepeatDatasetId: DatasetId!, $includeShortTandemRepeats: Boolean!) {
    region(chrom: $chrom, start: $start, stop: $stop, reference_genome: $referenceGenome) {
      genes {
        gene_id
        symbol
        start
        stop
        exons {
          feature_type
          start
          stop
        }
        transcripts {
          transcript_id
          exons {
            feature_type
            start
            stop
          }
        }
      }
      non_coding_constraints {
        start
        stop
        oe
        z
      }
      short_tandem_repeats(dataset: $shortTandemRepeatDatasetId) @include(if: $includeShortTandemRepeats) {
        id
      }
    }
  }
"
  success={[Function]}
  url="/api/"
  variables={
    {
      "chrom": "12",
      "includeShortTandemRepeats": false,
      "referenceGenome": "GRCh37",
      "shortTandemRepeatDatasetId": "gnomad_r3",
      "start": 345,
      "stop": 678,
    }
  }
>
  [Function]
</Query>
`;

exports[`RegionPageContainer with dataset gnomad_r2_1_non_neuro queries API with correct parameters 1`] = `
<Query
  errorMessage="Unable to load region"
  loadingMessage="Loading region"
  operationName="Region"
  query="
  query Region($chrom: String!, $start: Int!, $stop: Int!, $referenceGenome: ReferenceGenomeId!, $shortTandemRepeatDatasetId: DatasetId!, $includeShortTandemRepeats: Boolean!) {
    region(chrom: $chrom, start: $start, stop: $stop, reference_genome: $referenceGenome) {
      genes {
        gene_id
        symbol
        start
        stop
        exons {
          feature_type
          start
          stop
        }
        transcripts {
          transcript_id
          exons {
            feature_type
            start
            stop
          }
        }
      }
      non_coding_constraints {
        start
        stop
        oe
        z
      }
      short_tandem_repeats(dataset: $shortTandemRepeatDatasetId) @include(if: $includeShortTandemRepeats) {
        id
      }
    }
  }
"
  success={[Function]}
  url="/api/"
  variables={
    {
      "chrom": "12",
      "includeShortTandemRepeats": false,
      "referenceGenome": "GRCh37",
      "shortTandemRepeatDatasetId": "gnomad_r3",
      "start": 345,
      "stop": 678,
    }
  }
>
  [Function]
</Query>
`;

exports[`RegionPageContainer with dataset gnomad_r2_1_non_topmed queries API with correct parameters 1`] = `
<Query
  errorMessage="Unable to load region"
  loadingMessage="Loading region"
  operationName="Region"
  query="
  query Region($chrom: String!, $start: Int!, $stop: Int!, $referenceGenome: ReferenceGenomeId!, $shortTandemRepeatDatasetId: DatasetId!, $includeShortTandemRepeats: Boolean!) {
    region(chrom: $chrom, start: $start, stop: $stop, reference_genome: $referenceGenome) {
      genes {
        gene_id
        symbol
        start
        stop
        exons {
          feature_type
          start
          stop
        }
        transcripts {
          transcript_id
          exons {
            feature_type
            start
            stop
          }
        }
      }
      non_coding_constraints {
        start
        stop
        oe
        z
      }
      short_tandem_repeats(dataset: $shortTandemRepeatDatasetId) @include(if: $includeShortTandemRepeats) {
        id
      }
    }
  }
"
  success={[Function]}
  url="/api/"
  variables={
    {
      "chrom": "12",
      "includeShortTandemRepeats": false,
      "referenceGenome": "GRCh37",
      "shortTandemRepeatDatasetId": "gnomad_r3",
      "start": 345,
      "stop": 678,
    }
  }
>
  [Function]
</Query>
`;

exports[`RegionPageContainer with dataset gnomad_r3 queries API with correct parameters 1`] = `
<Query
  errorMessage="Unable to load region"
  loadingMessage="Loading region"
  operationName="Region"
  query="
  query Region($chrom: String!, $start: Int!, $stop: Int!, $referenceGenome: ReferenceGenomeId!, $shortTandemRepeatDatasetId: DatasetId!, $includeShortTandemRepeats: Boolean!) {
    region(chrom: $chrom, start: $start, stop: $stop, reference_genome: $referenceGenome) {
      genes {
        gene_id
        symbol
        start
        stop
        exons {
          feature_type
          start
          stop
        }
        transcripts {
          transcript_id
          exons {
            feature_type
            start
            stop
          }
        }
      }
      non_coding_constraints {
        start
        stop
        oe
        z
      }
      short_tandem_repeats(dataset: $shortTandemRepeatDatasetId) @include(if: $includeShortTandemRepeats) {
        id
      }
    }
  }
"
  success={[Function]}
  url="/api/"
  variables={
    {
      "chrom": "12",
      "includeShortTandemRepeats": true,
      "referenceGenome": "GRCh38",
      "shortTandemRepeatDatasetId": "gnomad_r3",
      "start": 345,
      "stop": 678,
    }
  }
>
  [Function]
</Query>
`;

exports[`RegionPageContainer with dataset gnomad_r3_controls_and_biobanks queries API with correct parameters 1`] = `
<Query
  errorMessage="Unable to load region"
  loadingMessage="Loading region"
  operationName="Region"
  query="
  query Region($chrom: String!, $start: Int!, $stop: Int!, $referenceGenome: ReferenceGenomeId!, $shortTandemRepeatDatasetId: DatasetId!, $includeShortTandemRepeats: Boolean!) {
    region(chrom: $chrom, start: $start, stop: $stop, reference_genome: $referenceGenome) {
      genes {
        gene_id
        symbol
        start
        stop
        exons {
          feature_type
          start
          stop
        }
        transcripts {
          transcript_id
          exons {
            feature_type
            start
            stop
          }
        }
      }
      non_coding_constraints {
        start
        stop
        oe
        z
      }
      short_tandem_repeats(dataset: $shortTandemRepeatDatasetId) @include(if: $includeShortTandemRepeats) {
        id
      }
    }
  }
"
  success={[Function]}
  url="/api/"
  variables={
    {
      "chrom": "12",
      "includeShortTandemRepeats": true,
      "referenceGenome": "GRCh38",
      "shortTandemRepeatDatasetId": "gnomad_r3",
      "start": 345,
      "stop": 678,
    }
  }
>
  [Function]
</Query>
`;

exports[`RegionPageContainer with dataset gnomad_r3_non_cancer queries API with correct parameters 1`] = `
<Query
  errorMessage="Unable to load region"
  loadingMessage="Loading region"
  operationName="Region"
  query="
  query Region($chrom: String!, $start: Int!, $stop: Int!, $referenceGenome: ReferenceGenomeId!, $shortTandemRepeatDatasetId: DatasetId!, $includeShortTandemRepeats: Boolean!) {
    region(chrom: $chrom, start: $start, stop: $stop, reference_genome: $referenceGenome) {
      genes {
        gene_id
        symbol
        start
        stop
        exons {
          feature_type
          start
          stop
        }
        transcripts {
          transcript_id
          exons {
            feature_type
            start
            stop
          }
        }
      }
      non_coding_constraints {
        start
        stop
        oe
        z
      }
      short_tandem_repeats(dataset: $shortTandemRepeatDatasetId) @include(if: $includeShortTandemRepeats) {
        id
      }
    }
  }
"
  success={[Function]}
  url="/api/"
  variables={
    {
      "chrom": "12",
      "includeShortTandemRepeats": true,
      "referenceGenome": "GRCh38",
      "shortTandemRepeatDatasetId": "gnomad_r3",
      "start": 345,
      "stop": 678,
    }
  }
>
  [Function]
</Query>
`;

exports[`RegionPageContainer with dataset gnomad_r3_non_neuro queries API with correct parameters 1`] = `
<Query
  errorMessage="Unable to load region"
  loadingMessage="Loading region"
  operationName="Region"
  query="
  query Region($chrom: String!, $start: Int!, $stop: Int!, $referenceGenome: ReferenceGenomeId!, $shortTandemRepeatDatasetId: DatasetId!, $includeShortTandemRepeats: Boolean!) {
    region(chrom: $chrom, start: $start, stop: $stop, reference_genome: $referenceGenome) {
      genes {
        gene_id
        symbol
        start
        stop
        exons {
          feature_type
          start
          stop
        }
        transcripts {
          transcript_id
          exons {
            feature_type
            start
            stop
          }
        }
      }
      non_coding_constraints {
        start
        stop
        oe
        z
      }
      short_tandem_repeats(dataset: $shortTandemRepeatDatasetId) @include(if: $includeShortTandemRepeats) {
        id
      }
    }
  }
"
  success={[Function]}
  url="/api/"
  variables={
    {
      "chrom": "12",
      "includeShortTandemRepeats": true,
      "referenceGenome": "GRCh38",
      "shortTandemRepeatDatasetId": "gnomad_r3",
      "start": 345,
      "stop": 678,
    }
  }
>
  [Function]
</Query>
`;

exports[`RegionPageContainer with dataset gnomad_r3_non_topmed queries API with correct parameters 1`] = `
<Query
  errorMessage="Unable to load region"
  loadingMessage="Loading region"
  operationName="Region"
  query="
  query Region($chrom: String!, $start: Int!, $stop: Int!, $referenceGenome: ReferenceGenomeId!, $shortTandemRepeatDatasetId: DatasetId!, $includeShortTandemRepeats: Boolean!) {
    region(chrom: $chrom, start: $start, stop: $stop, reference_genome: $referenceGenome) {
      genes {
        gene_id
        symbol
        start
        stop
        exons {
          feature_type
          start
          stop
        }
        transcripts {
          transcript_id
          exons {
            feature_type
            start
            stop
          }
        }
      }
      non_coding_constraints {
        start
        stop
        oe
        z
      }
      short_tandem_repeats(dataset: $shortTandemRepeatDatasetId) @include(if: $includeShortTandemRepeats) {
        id
      }
    }
  }
"
  success={[Function]}
  url="/api/"
  variables={
    {
      "chrom": "12",
      "includeShortTandemRepeats": true,
      "referenceGenome": "GRCh38",
      "shortTandemRepeatDatasetId": "gnomad_r3",
      "start": 345,
      "stop": 678,
    }
  }
>
  [Function]
</Query>
`;

exports[`RegionPageContainer with dataset gnomad_r3_non_v2 queries API with correct parameters 1`] = `
<Query
  errorMessage="Unable to load region"
  loadingMessage="Loading region"
  operationName="Region"
  query="
  query Region($chrom: String!, $start: Int!, $stop: Int!, $referenceGenome: ReferenceGenomeId!, $shortTandemRepeatDatasetId: DatasetId!, $includeShortTandemRepeats: Boolean!) {
    region(chrom: $chrom, start: $start, stop: $stop, reference_genome: $referenceGenome) {
      genes {
        gene_id
        symbol
        start
        stop
        exons {
          feature_type
          start
          stop
        }
        transcripts {
          transcript_id
          exons {
            feature_type
            start
            stop
          }
        }
      }
      non_coding_constraints {
        start
        stop
        oe
        z
      }
      short_tandem_repeats(dataset: $shortTandemRepeatDatasetId) @include(if: $includeShortTandemRepeats) {
        id
      }
    }
  }
"
  success={[Function]}
  url="/api/"
  variables={
    {
      "chrom": "12",
      "includeShortTandemRepeats": true,
      "referenceGenome": "GRCh38",
      "shortTandemRepeatDatasetId": "gnomad_r3",
      "start": 345,
      "stop": 678,
    }
  }
>
  [Function]
</Query>
`;

exports[`RegionPageContainer with dataset gnomad_sv_r2_1 queries API with correct parameters 1`] = `
<Query
  errorMessage="Unable to load region"
  loadingMessage="Loading region"
  operationName="Region"
  query="
  query Region($chrom: String!, $start: Int!, $stop: Int!, $referenceGenome: ReferenceGenomeId!, $shortTandemRepeatDatasetId: DatasetId!, $includeShortTandemRepeats: Boolean!) {
    region(chrom: $chrom, start: $start, stop: $stop, reference_genome: $referenceGenome) {
      genes {
        gene_id
        symbol
        start
        stop
        exons {
          feature_type
          start
          stop
        }
        transcripts {
          transcript_id
          exons {
            feature_type
            start
            stop
          }
        }
      }
      non_coding_constraints {
        start
        stop
        oe
        z
      }
      short_tandem_repeats(dataset: $shortTandemRepeatDatasetId) @include(if: $includeShortTandemRepeats) {
        id
      }
    }
  }
"
  success={[Function]}
  url="/api/"
  variables={
    {
      "chrom": "12",
      "includeShortTandemRepeats": false,
      "referenceGenome": "GRCh37",
      "shortTandemRepeatDatasetId": "gnomad_r3",
      "start": 345,
      "stop": 678,
    }
  }
>
  [Function]
</Query>
`;

exports[`RegionPageContainer with dataset gnomad_sv_r2_1_controls queries API with correct parameters 1`] = `
<Query
  errorMessage="Unable to load region"
  loadingMessage="Loading region"
  operationName="Region"
  query="
  query Region($chrom: String!, $start: Int!, $stop: Int!, $referenceGenome: ReferenceGenomeId!, $shortTandemRepeatDatasetId: DatasetId!, $includeShortTandemRepeats: Boolean!) {
    region(chrom: $chrom, start: $start, stop: $stop, reference_genome: $referenceGenome) {
      genes {
        gene_id
        symbol
        start
        stop
        exons {
          feature_type
          start
          stop
        }
        transcripts {
          transcript_id
          exons {
            feature_type
            start
            stop
          }
        }
      }
      non_coding_constraints {
        start
        stop
        oe
        z
      }
      short_tandem_repeats(dataset: $shortTandemRepeatDatasetId) @include(if: $includeShortTandemRepeats) {
        id
      }
    }
  }
"
  success={[Function]}
  url="/api/"
  variables={
    {
      "chrom": "12",
      "includeShortTandemRepeats": false,
      "referenceGenome": "GRCh37",
      "shortTandemRepeatDatasetId": "gnomad_r3",
      "start": 345,
      "stop": 678,
    }
  }
>
  [Function]
</Query>
`;

exports[`RegionPageContainer with dataset gnomad_sv_r2_1_non_neuro queries API with correct parameters 1`] = `
<Query
  errorMessage="Unable to load region"
  loadingMessage="Loading region"
  operationName="Region"
  query="
  query Region($chrom: String!, $start: Int!, $stop: Int!, $referenceGenome: ReferenceGenomeId!, $shortTandemRepeatDatasetId: DatasetId!, $includeShortTandemRepeats: Boolean!) {
    region(chrom: $chrom, start: $start, stop: $stop, reference_genome: $referenceGenome) {
      genes {
        gene_id
        symbol
        start
        stop
        exons {
          feature_type
          start
          stop
        }
        transcripts {
          transcript_id
          exons {
            feature_type
            start
            stop
          }
        }
      }
      non_coding_constraints {
        start
        stop
        oe
        z
      }
      short_tandem_repeats(dataset: $shortTandemRepeatDatasetId) @include(if: $includeShortTandemRepeats) {
        id
      }
    }
  }
"
  success={[Function]}
  url="/api/"
  variables={
    {
      "chrom": "12",
      "includeShortTandemRepeats": false,
      "referenceGenome": "GRCh37",
      "shortTandemRepeatDatasetId": "gnomad_r3",
      "start": 345,
      "stop": 678,
    }
  }
>
  [Function]
</Query>
`;

exports[`RegionPageContainer with dataset gnomad_sv_r4 queries API with correct parameters 1`] = `
<Query
  errorMessage="Unable to load region"
  loadingMessage="Loading region"
  operationName="Region"
  query="
  query Region($chrom: String!, $start: Int!, $stop: Int!, $referenceGenome: ReferenceGenomeId!, $shortTandemRepeatDatasetId: DatasetId!, $includeShortTandemRepeats: Boolean!) {
    region(chrom: $chrom, start: $start, stop: $stop, reference_genome: $referenceGenome) {
      genes {
        gene_id
        symbol
        start
        stop
        exons {
          feature_type
          start
          stop
        }
        transcripts {
          transcript_id
          exons {
            feature_type
            start
            stop
          }
        }
      }
      non_coding_constraints {
        start
        stop
        oe
        z
      }
      short_tandem_repeats(dataset: $shortTandemRepeatDatasetId) @include(if: $includeShortTandemRepeats) {
        id
      }
    }
  }
"
  success={[Function]}
  url="/api/"
  variables={
    {
      "chrom": "12",
      "includeShortTandemRepeats": true,
      "referenceGenome": "GRCh38",
      "shortTandemRepeatDatasetId": "gnomad_r3",
      "start": 345,
      "stop": 678,
    }
  }
>
  [Function]
</Query>
`;
